frontend/src/metabase/visualizations/visualizations/SmartScalar/SmartScalar.jsx

```diff
@@ -284,12 +289,20 @@ Object.assign(SmartScalar, {
       section: t`Data`,
       title: t`Comparisons`,
       widget: SmartScalarComparisonWidget,
-      isValid: (series, vizSettings) => isComparisonValid(series, vizSettings),
-      getDefault: (series, vizSettings) =>
-        getDefaultComparison(series, vizSettings),
+      isValid: (series, vizSettings) => {
+        const comparisons = vizSettings["scalar.comparisons"];
+        return comparisons.every(comparison =>
+          isComparisonValid(comparison, series, vizSettings),
```

Comment:
I would iterate inside `isComparisonValid` function and rename it. The same is about `getDefaultComparison` which can return an array right away. The reason is that we will reuse it as is in the static viz land so we won't duplicate the iteration part.

Also `getDefaultComparison` can be improved to return all valid comparisons if for instance there are 3 but only 1 is invalid